diff --git a/src/xrt/drivers/CMakeLists.txt b/src/xrt/drivers/CMakeLists.txt
index c8adf727a..048aee9b2 100644
--- a/src/xrt/drivers/CMakeLists.txt
+++ b/src/xrt/drivers/CMakeLists.txt
@@ -296,6 +296,8 @@ if(XRT_BUILD_DRIVER_ROKID)
 		drv_rokid
 		PRIVATE
 			xrt-interfaces
+			tracking-includes
+			constellation
 			aux_util
 			aux_os
 			aux_math
@@ -562,6 +564,8 @@ if(XRT_BUILD_DRIVER_WMR)
 		drv_wmr
 		PRIVATE
 			xrt-interfaces
+			tracking-includes
+			constellation
 			aux_util
 			aux_util_sink # TODO remove this when converted to builders.
 			aux_math
